src/language/variables.md

```diff
@@ -38,6 +38,48 @@ String name = 'Bob';
   of using `var`, rather than type annotations, for local variables.
 {{site.alert.end}}
 
+## Null safety
+
+The Dart language enforces sound null safety.
+
+Null safety prevents an error that results from unintentional access
+of variables set to `null`. The error is called a null dereference error.
+If you try to access a property or call a method on a variable set to `null`,
+that causes a null dereference error.
+For example, say you want to find the absolute value of an `int` variable `i`.
+If `i` is `null`, calling `i.abs()` causes a null dereference error.
+In other languages, trying this could lead to a runtime error,
+but Dart's compiler prohibits these actions.
+Therefore, Dart apps can't cause runtime errors.
+
+Null safety introduces two key changes to type declaration.
+
+First, you can specify that a variable _can_ be set to `null`.
+To call out nullability, you add a `?` to the end of the type declaration.
+
+```dart
+String? name  // Nullable type. Can be `null` or string.
+
+String name   // Default type. Cannot be `null` but can be string.
+```
+
+Second, you must initialize variables before using them.
```

Comment:
This is a key change, but it's not a change to type declarations/annotations. I'd probably say:

---

Null safety introduces two key changes.

First, when specifying a type for a variable, parameter, return type, etc., you can control whether the type allows `null`. ...

Second, you must initialize variables before using them.

---

There is a third key change to call out here: You can't access properties or call methods on an expression whose type is nullable (unless it's a property or method that's defined on `Null` like `hashCode` or `toString()`).